File: android/app/src/main/java/com/celalbasaran/stripmate/data/model/Achievement.kt
package com.celalbasaran.stripmate.data.model

import java.util.Date

data class Achievement(
    val id: String,
    val title: String,
    val description: String,
    val emoji: String,
    val category: AchievementCategory,
    val requirement: Int
) {
    companion object {
        val ALL_ACHIEVEMENTS: List<Achievement> = listOf(
            // Photo milestones
            Achievement("first_photo", "ilk an", "ilk fotoğrafını gönder", "\uD83D\uDCF8", AchievementCategory.PHOTOS, 1),
            Achievement("photos_10", "anları biriktiren", "10 fotoğraf gönder", "\uD83C\uDF9E\uFE0F", AchievementCategory.PHOTOS, 10),
            Achievement("photos_50", "fotoğraf tutkunu", "50 fotoğraf gönder", "\uD83C\uDF1F", AchievementCategory.PHOTOS, 50),
            Achievement("photos_100", "yüz an", "100 fotoğraf gönder", "\uD83D\uDC8E", AchievementCategory.PHOTOS, 100),
            Achievement("photos_500", "efsane", "500 fotoğraf gönder", "\uD83D\uDC51", AchievementCategory.PHOTOS, 500),

            // Streak milestones
            Achievement("streak_7", "bir hafta", "7 günlük seri yakala", "\uD83D\uDD25", AchievementCategory.STREAKS, 7),
            Achievement("streak_30", "bir ay", "30 günlük seri yakala", "\u26A1", AchievementCategory.STREAKS, 30),
            Achievement("streak_100", "yüz gün", "100 günlük seri yakala", "\uD83C\uDFC6", AchievementCategory.STREAKS, 100),
            Achievement("streak_365", "bir yıl", "365 günlük seri yakala", "\uD83D\uDCAB", AchievementCategory.STREAKS, 365),

            // Social milestones
            Achievement("first_friend", "ilk bağlantı", "ilk arkadaşını ekle", "\uD83E\uDD1D", AchievementCategory.SOCIAL, 1),
            Achievement("friends_5", "beşli", "5 arkadaş edin", "\uD83D\uDC65", AchievementCategory.SOCIAL, 5),
            Achievement("friends_10", "popüler", "10 arkadaş edin", "\uD83C\uDF10", AchievementCategory.SOCIAL, 10),
            Achievement("friends_25", "sosyal kelebek", "25 arkadaş edin", "\uD83E\uDD8B", AchievementCategory.SOCIAL, 25),
            Achievement("first_comment", "ilk yorum", "ilk yorumunu yaz", "\uD83D\uDCAC", AchievementCategory.SOCIAL, 1),
            Achievement("reaction_50", "tepki makinesi", "50 reaksiyon ver", "\uD83C\uDFAD", AchievementCategory.SOCIAL, 50),
            Achievement("dm_100", "sohbet ustası", "100 DM gönder", "\u2709\uFE0F", AchievementCategory.SOCIAL, 100),

            // Explorer milestones
            Achievement("cities_3", "gezgin", "3 farklı şehirden fotoğraf gönder", "\uD83D\uDDFA\uFE0F", AchievementCategory.EXPLORER, 3),
            Achievement("cities_10", "kaşif", "10 farklı şehirden fotoğraf gönder", "\uD83E\uDDED", AchievementCategory.EXPLORER, 10),
            Achievement("daily_prompt_7", "görev canavarı", "7 günlük görev tamamla", "\u2705", AchievementCategory.EXPLORER, 7),
            Achievement("daily_prompt_30", "görev ustası", "30 günlük görev tamamla", "\uD83C\uDFAF", AchievementCategory.EXPLORER, 30),
            Achievement("night_owl", "gece kuşu", "gece yarısından sonra fotoğraf gönder", "\uD83E\uDD89", AchievementCategory.EXPLORER, 1),
            Achievement("early_bird", "erken kuş", "sabah 7'den önce fotoğraf gönder", "\uD83D\uDC26", AchievementCategory.EXPLORER, 1),
            Achievement("memory_lane", "anı yolu", "'Bugün Geçen Yıl' anısını görüntüle", "\uD83D\uDD70\uFE0F", AchievementCategory.EXPLORER, 1)
        )

        fun findById(id: String): Achievement? =
            ALL_ACHIEVEMENTS.firstOrNull { it.id == id }
    }
}

enum class AchievementCategory(val displayName: String) {
    PHOTOS("fotoğraf"),
    STREAKS("seri"),
    SOCIAL("sosyal"),
    EXPLORER("kaşif")
}

data class UserAchievement(
    val achievementId: String = "",
    val unlockedAt: Date = Date()
)

File: StripMate/Models/Achievement.swift
import Foundation

/// Achievement badge definitions for anlık.
public struct Achievement: Identifiable, Codable, Sendable {
    public let id: String
    public let title: String
    public let description: String
    public let emoji: String
    public let category: Category
    public let requirement: Int  // threshold to unlock
    
    public enum Category: String, Codable, Sendable, CaseIterable {
        case photos = "fotoğraf"
        case streaks = "seri"
        case social = "sosyal"
        case explorer = "kaşif"
    }
    
    // All available achievements
    public static let all: [Achievement] = [
        // Photo milestones
        Achievement(id: "first_photo", title: String(localized: "ilk an"), description: String(localized: "ilk fotoğrafını gönder"), emoji: "camera.fill", category: .photos, requirement: 1),
        Achievement(id: "photos_10", title: String(localized: "anları biriktiren"), description: String(localized: "10 fotoğraf gönder"), emoji: "film", category: .photos, requirement: 10),
        Achievement(id: "photos_50", title: String(localized: "fotoğraf tutkunu"), description: String(localized: "50 fotoğraf gönder"), emoji: "star.fill", category: .photos, requirement: 50),
        Achievement(id: "photos_100", title: String(localized: "yüz an"), description: String(localized: "100 fotoğraf gönder"), emoji: "diamond.fill", category: .photos, requirement: 100),
        Achievement(id: "photos_500", title: String(localized: "efsane"), description: String(localized: "500 fotoğraf gönder"), emoji: "crown.fill", category: .photos, requirement: 500),

        // Streak milestones
        Achievement(id: "streak_7", title: String(localized: "bir hafta"), description: String(localized: "7 günlük bağ yakala"), emoji: "flame.fill", category: .streaks, requirement: 7),
        Achievement(id: "streak_30", title: String(localized: "bir ay"), description: String(localized: "30 günlük bağ yakala"), emoji: "bolt.fill", category: .streaks, requirement: 30),
        Achievement(id: "streak_100", title: String(localized: "yüz gün"), description: String(localized: "100 günlük bağ yakala"), emoji: "trophy.fill", category: .streaks, requirement: 100),
        Achievement(id: "streak_365", title: String(localized: "bir yıl"), description: String(localized: "365 günlük bağ yakala"), emoji: "sparkles", category: .streaks, requirement: 365),

        // Social milestones
        Achievement(id: "first_friend", title: String(localized: "ilk bağlantı"), description: String(localized: "ilk arkadaşını ekle"), emoji: "person.2.fill", category: .social, requirement: 1),
        Achievement(id: "friends_5", title: String(localized: "beşli"), description: String(localized: "5 arkadaş edin"), emoji: "person.3.fill", category: .social, requirement: 5),
        Achievement(id: "friends_10", title: String(localized: "popüler"), description: String(localized: "10 arkadaş edin"), emoji: "globe", category: .social, requirement: 10),
        Achievement(id: "friends_25", title: String(localized: "sosyal kelebek"), description: String(localized: "25 arkadaş edin"), emoji: "leaf.fill", category: .social, requirement: 25),
        Achievement(id: "first_comment", title: String(localized: "ilk yorum"), description: String(localized: "ilk yorumunu yaz"), emoji: "bubble.left.fill", category: .social, requirement: 1),
        Achievement(id: "reaction_50", title: String(localized: "tepki makinesi"), description: String(localized: "50 reaksiyon ver"), emoji: "theatermasks.fill", category: .social, requirement: 50),
        Achievement(id: "dm_100", title: String(localized: "sohbet ustası"), description: String(localized: "100 DM gönder"), emoji: "envelope.fill", category: .social, requirement: 100),

        // Explorer milestones
        Achievement(id: "cities_3", title: String(localized: "gezgin"), description: String(localized: "3 farklı şehirden fotoğraf gönder"), emoji: "map.fill", category: .explorer, requirement: 3),
        Achievement(id: "cities_10", title: String(localized: "kaşif"), description: String(localized: "10 farklı şehirden fotoğraf gönder"), emoji: "safari.fill", category: .explorer, requirement: 10),
        Achievement(id: "daily_prompt_7", title: String(localized: "görev canavarı"), description: String(localized: "7 günlük görev tamamla"), emoji: "checkmark.seal.fill", category: .explorer, requirement: 7),
        Achievement(id: "daily_prompt_30", title: String(localized: "görev ustası"), description: String(localized: "30 günlük görev tamamla"), emoji: "target", category: .explorer, requirement: 30),
        Achievement(id: "night_owl", title: String(localized: "gece kuşu"), description: String(localized: "gece yarısından sonra fotoğraf gönder"), emoji: "moon.fill", category: .explorer, requirement: 1),
        Achievement(id: "early_bird", title: String(localized: "erken kuş"), description: String(localized: "sabah 7'den önce fotoğraf gönder"), emoji: "sunrise.fill", category: .explorer, requirement: 1),
        Achievement(id: "memory_lane", title: String(localized: "anı yolu"), description: String(localized: "'Bugün Geçen Yıl' anısını görüntüle"), emoji: "clock.fill", category: .explorer, requirement: 1),
    ]
}

/// User's unlocked achievements
public struct UserAchievement: Codable, Sendable {
    public let achievementId: String
    public let unlockedAt: Date
}
